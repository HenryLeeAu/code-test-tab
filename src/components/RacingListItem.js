import React from 'react';
import timeToGo from 'tools/timeToGo'
export default (props) =>{
  const {raceStartTime,raceName,raceNumber,meeting}=props.raceItem;
  const {meetingDate,meetingName,location} = meeting;

  return (
    <li>
      <button>
        <div className="raceNumber"><strong>R{raceNumber}</strong></div>
        <div><h2 className="raceName">{raceName}</h2></div>
        <div>{timeToGo(props.currentTime,raceStartTime)}</div>
        <div className="meetingName">{meetingName}({location})</div>
        <div className="meetingDate">{meetingDate}</div>
      </button>
    </li>
  )
}