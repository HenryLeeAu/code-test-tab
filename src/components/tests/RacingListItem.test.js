import React from 'react';
import RacingListItem from 'components/RacingListItem';
import { shallow } from 'enzyme';
let wrappered;
const props = {
  currentTime:"2019-01-16T03:28:00.000Z",
  raceItem:{
    raceStartTime:"2019-01-16T03:29:00.000Z",
    raceName: "HATRICK RACE 15",
    raceNumbe: 5,
    meeting:{
      meetingName: "NEW ZEALAND",
      location: "NZL",
      meetingDate: "2019-01-16"
    }
  }
}
describe('RacingListItem', () => {
  
  it('render UI ', () => {
    wrappered = shallow(<RacingListItem {...props} />);
    expect(wrappered).toMatchSnapshot();
  });
});
