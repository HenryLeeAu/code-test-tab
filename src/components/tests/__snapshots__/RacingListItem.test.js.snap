// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RacingListItem render UI  1`] = `
<li>
  <button>
    <div
      className="raceNumber"
    >
      <strong>
        R
      </strong>
    </div>
    <div>
      <h2
        className="raceName"
      >
        HATRICK RACE 15
      </h2>
    </div>
    <div>
        1M
    </div>
    <div
      className="meetingName"
    >
      NEW ZEALAND
      (
      NZL
      )
    </div>
    <div
      className="meetingDate"
    >
      2019-01-16
    </div>
  </button>
</li>
`;
